note.py

```diff
@@ -75,20 +95,33 @@ def remove_backlink(note_name, backlink):
     note = get_note(note_name)
     try:
         note.backlinks.remove(backlink)
+        note_dict = note.dict()
+        note_content = drive_notes.get(note_name)
+        if note_content:
+            note_content = str(note_content.read(), 'ascii')
+            note_dict["content"] = note_content
+        else:
+            note_dict["content"] = default_content
     except ValueError:
         pass
-    return db_update_note(note)
-    
+    return db_update_note(Note(**note_dict))
+
 
 # add a backlink to a new note, if non-existent, create
 def add_backlink_or_create(note_name, backlink):
-      note = get_note(note_name)
-      
-      if note:
-        note.backlinks.append(backlink)
-        return db_update_note(note)
+    note = get_note(note_name)
 
-      else:
+    if note:
```

Comment:
For this function, if it's only a backlink update, we don't need to read from drive (because the content itself doesn't change).

Maybe a slight refactor of db_update_note?

Or maybe 3 functions:
1. `update_note`: calls 2 & 3
2. `drive_update_note`: reads / writes from Drive
3. `db_update_note`: reads / writes from Base